src/engine/combiners/and.js

```diff
@@ -17,27 +17,37 @@ limitations under the License.
 
 'use strict';
 
+const findIndex = require('core-js/library/fn/array/find-index');
 const log = require('debug')('splitio-engine:combiner');
 
-function andCombinerContext(matchers: Array<Function>): Function {
+function andResults(results) {
+  let i = 0;
+  let len = results.length;
+  let hasMatchedAll;
+
+  // loop through all the matchers an stop at the first one returning false.
+  for (; i < len && results[i]; i++) {
+    // logic is run inside the condition of evaluates next step.
```

Comment:
Given a boolean array [ true, true, true, ... ] I'm computing true if and only if all the values are true.